telluric/rasterization.py

```diff
@@ -1,28 +1,47 @@
+import warnings
+
 import numpy as np
 
 from affine import Affine, TransformNotInvertibleError
 from rasterio import features
+from rasterio.dtypes import get_minimum_dtype
 from rasterio._err import CPLE_OutOfMemoryError  # noqa
 
 from telluric.georaster import GeoRaster2
 
 
-DTYPE = np.uint8
 NODATA_VALUE = 0
-# FILL_VALUE = np.iinfo(DTYPE).max
+# FILL_VALUE = np.iinfo(np.uint8).max
 FILL_VALUE = 1
 
+NODATA_DEPRECATION_WARNING = ("Passing nodata_value to rasterize is not supported anymore. "
+                              "An appropriate nodata value will be generated, depending on the fill value(s).")
+
 
 class ScaleError(ValueError):
     pass
 
 
-def rasterize(shapes, crs, bounds, dest_resolution, fill_value=None, nodata_value=None,
-              band_names=None, dtype=np.uint8):
+def rasterize(shapes, crs, bounds, dest_resolution, *, fill_value=None,
```

Comment:
why did you add the `*`?

Reply:
To make sure the user does not give a `nodata_value` by accident and provide a good error message.

Reply:
ok